src/Microsoft.PowerShell.Commands.Management/commands/management/Navigation.cs

```diff
@@ -2698,8 +2698,9 @@ protected override void ProcessRecord()
                     {
                         try
                         {
-                            System.IO.DirectoryInfo di = new(providerPath);
-                            if (di != null && InternalSymbolicLinkLinkCodeMethods.IsReparsePointLikeSymlink(di))
+                            var directoryInfo = new System.IO.DirectoryInfo(providerPath);
+                            if (directoryInfo.Exists
+                                && InternalSymbolicLinkLinkCodeMethods.IsReparsePointLikeSymlink(directoryInfo))
```

Comment:
Here `di` is never null and we can remove the null check but we also have no need to do extra Exist check because `providerPath` is based on `resolvedPath`.